src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java

```diff
@@ -565,6 +592,39 @@ static SplitLocusRecursive splitLocusIntervals1(LinkedList<GenomeLoc> remaining,
         return new SplitLocusRecursive(split, remaining);
     }
 
+    /**
+     * Check whether to locatable overlap.
+     * <p>
+     *    Two locatables overlap if the share the same contig and they have at least one
+     *    base in common based on their start and end positions.
+     * </p>
+     * <p>
+     *    This method returns {@code false} if either input {@link Locatable} has a {@code null}
+     *    contig.
+     * </p>
+     *
+     * @param left first locatable.
+     * @param right second locatable.
+     * @throws IllegalArgumentException if either {@code left} or {@code right} locatable
+     *  is {@code null}.
+     * @return {@code true} iff there is an overlap between both locatables.
+     */
+    public static boolean overlaps(final Locatable left,final Locatable right) {
+        Utils.nonNull(left,"the left locatable is null");
+        Utils.nonNull(right,"the right locatable is null");
+        if (left.getContig() == null || right.getContig() == null) {
+            return false;
+        } else if (!left.getContig().equals(right.getContig())) {
+            return false;
+        } else {
+            // If there is no overlap the cross-differences between
+            // start and ends should both either be positive or negative;
+            // never mixed or zero.
+            return Integer.compare(left.getStart(),right.getEnd()) *
+                    Integer.compare(left.getEnd(),right.getStart()) <= 0;
```

Comment:
Prefer the much less confusing (and cheaper) `return left.getStart() <= right.getEnd() && right.getStart() <= left.getEnd();`